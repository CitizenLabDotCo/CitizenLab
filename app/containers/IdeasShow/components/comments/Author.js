import React from 'react';
import { FormattedRelative, FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import styled from 'styled-components';
import { Link } from 'react-router';

import { preprocess } from 'utils';

// import IdeaContent from '../IdeaContent';
import { selectResourcesDomain } from 'utils/resources/selectors';

const AuthorContainer = styled.div`
  display: flex;
  align-items: center;
`;

const Avatar = styled.img`
  flex: 0 0 38px;
  height: 38px;
  border-radius: 50%;
`;

const AuthorName = styled(Link)`
  font-weight: bold;
  color: #484848;
  font-size: 16px;
  flex-grow: 1;
  padding-left: 12px;
`;

const Timing = styled.div`
  font-size: 14px;
  color: #a9a9a9;
`;

/* eslint-disable */
const Author = ({id, slug, firstName, lastName, avatar, createdAt, message}) => (
  <AuthorContainer>
    <Avatar src={avatar} />
    <AuthorName to={`/profile/${slug}`}>
      <FormattedMessage {...message} values={{ firstName, lastName }} />
    </AuthorName>
    <Timing>
      <FormattedRelative value={createdAt} />
    </Timing>
  </AuthorContainer>
);

Author.propTypes = {
  first_name: PropTypes.string,
  last_name: PropTypes.string,
  avatar: PropTypes.string,
  createdAt: PropTypes.any,
};

const mapStateToProps = () => createStructuredSelector({
  user: (state, { authorId }) => selectResourcesDomain('users', authorId)(state),
});

const mergeProps = ({ user }, dispatchProps, ownProps) => {
  const { createdAt, message } = ownProps;
  if (!user) return {}
  const attributes = user.get('attributes');
  const id = user.get('id');
  const slug = attributes.get('slug');
  const firstName = attributes.get('first_name');
  const lastName = attributes.get('last_name');
  const avatar = attributes.getIn(['avatar', 'small']);

  return {
    id,
    slug,
    avatar,
    firstName,
    lastName,
    lastName,
    createdAt,
    message,
  };

};


export default preprocess(mapStateToProps, null, mergeProps)(Author)
