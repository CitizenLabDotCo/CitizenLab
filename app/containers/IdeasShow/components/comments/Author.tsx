import * as React from 'react';
import * as _ from 'lodash';
import * as Rx from 'rxjs/Rx';

// components
import { Link } from 'react-router';
import Avatar from 'components/Avatar';

// services
import { userStream, IUser } from 'services/users';

// i18n
import { FormattedRelative, FormattedMessage } from 'react-intl';
import messages from '../../messages';

// style
import styled from 'styled-components';

const AuthorContainer = styled.div`
  display: flex;
  align-items: center;
`;

const AuthorAvatar = styled(Avatar)`
  flex: 0 0 30px;
  height: 30px;
  width: 30px;
`;

const AuthorName = styled(Link)`
  font-weight: bold;
  color: #484848;
  font-size: 16px;
  flex-grow: 1;
  padding-left: 12px;
`;

const Timing = styled.div`
  font-size: 14px;
  color: #a9a9a9;
`;

type Props = {
  authorId: string;
  createdAt: string;
};

type State = {
  author: IUser | null;
};

export default class Author extends React.PureComponent<Props, State> {
  state: State;
  subscriptions: Rx.Subscription[];

  constructor() {
    super();
    this.state = {
      author: null
    };
    this.subscriptions = [];
  }

  componentWillMount() {
    const { authorId } = this.props;
    const author$ = userStream(authorId).observable;

    this.subscriptions = [
      author$.subscribe(author => this.setState({ author }))
    ];
  }

  componentWillUnmount() {
    this.subscriptions.forEach(subscription => subscription.unsubscribe());
  }

  render() {
    const { authorId, createdAt } = this.props;
    const { author } = this.state;

    if (author) {
      const avatar = author.data.attributes.avatar.medium;
      const slug = author.data.attributes.slug;
      const firstName = author.data.attributes.first_name;
      const lastName = author.data.attributes.last_name;

      return (
        <AuthorContainer>
          <AuthorAvatar userId={authorId} size="medium" />

          <AuthorName to={`/profile/${slug}`}>
            <FormattedMessage {...messages.authorSaid} values={{ firstName }} />
          </AuthorName>

          <Timing>
            <FormattedRelative value={createdAt} />
          </Timing>
        </AuthorContainer>
      );
    }

    return null;
  }
}
