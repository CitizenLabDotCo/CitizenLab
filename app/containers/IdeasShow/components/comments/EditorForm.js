import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
import { convertToRaw } from 'draft-js';
import draftToHtml from 'draftjs-to-html';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { makeSelectLocale } from 'containers/LanguageProvider/selectors';


// components
import Button from 'components/UI/Button';
import Editor from 'components/UI/Editor';

// messages
import messages from '../../messages';
import { publishCommentRequest } from '../../actions';
import selectIdeasShow from '../../selectors';

const SubmitButton = styled(Button)`
  float: right;
  margin-top: 20px;
`;

class EditorForm extends React.PureComponent {
  constructor(props) {
    super(props);
    this.values = { ideaId: props.ideaId, parentId: props.parentId };
    this.state = {
      editorState: null,
    };
  }

  handleEditorChange = (editorState) => {
    this.setState({ editorState });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const editorContent = convertToRaw(this.state.editorState.getCurrentContent());
    const htmlContent = draftToHtml(editorContent);
    if (htmlContent && htmlContent.trim() !== '<p></p>') {
      const bodyMultiloc = {};
      bodyMultiloc[this.props.locale] = htmlContent;

      const comment = {
        parent_id: this.props.parentId,
        body_multiloc: bodyMultiloc,
      };
      this.props.publishCommentRequest(this.props.ideaId, comment);
    }
  }

  /* eslint-disable react/jsx-boolean-value*/
  render() {
    const { formatMessage } = this.props.intl;
    const { formStatus, error } = this.props;
    if (formStatus !== 'success') {
      return (
        <form onSubmit={this.handleSubmit}>
          <Editor
            id="editor"
            value={this.state.editorState}
            placeholder={formatMessage(messages.commentBodyPlaceholder)}
            onChange={this.handleEditorChange}
          />
          {formStatus === 'error' && <div>{error}</div>}
          <SubmitButton
            loading={formStatus === 'processing'}
          >
            <FormattedMessage {...messages.publishComment} />
          </SubmitButton>
          <div style={{ clear: 'both' }}></div>
        </form>
      );
    }
    // if formStatus === 'success'
    return (
      <div>
        <FormattedMessage {...messages.commentSuccess} />
      </div>
    );
  }

}

EditorForm.propTypes = {
  parentId: PropTypes.string,
  ideaId: PropTypes.string.isRequired,
  intl: intlShape.isRequired,
  publishCommentRequest: PropTypes.func,
  locale: PropTypes.string,
  formStatus: PropTypes.string, // undefined, 'processing', 'error' or 'success'
  error: PropTypes.string,
};

const mapDispatchToProps = {
  publishCommentRequest,
};

const mapStateToProps = (state, { parentId }) => ({
  locale: makeSelectLocale()(state),
  formStatus: selectIdeasShow('newComment', parentId || 'root', 'formStatus')(state),
  error: selectIdeasShow('newComment', parentId || 'root', 'error')(state),
});

export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(EditorForm));
