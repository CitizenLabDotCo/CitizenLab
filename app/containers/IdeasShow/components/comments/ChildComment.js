import React from 'react';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import styled from 'styled-components';

import { preprocess } from 'utils';
import { selectResourcesDomain } from 'utils/resources/selectors';
import T from 'containers/T';


// import MapChildren from './MapChildren';
import messages from '../../messages';
import Author from './Author';

const CommentContainer = styled.div`
  border: solid 1px #cdcdcd;
  padding: 25px;
  border-top: none;
`;

const CommentBody = styled.div`
  font-size: 16px;
  color: #8f8f8f;
  padding: 15px 0 15px 50px;
`;


class ChildComment extends React.Component {

  render() {
    const { content, createdAt, authorId } = this.props;
    return (
      <div>
        <CommentContainer>
          <Author authorId={authorId} createdAt={createdAt} message={messages.authorReacted} />
          <CommentBody>
            <T value={content} />
          </CommentBody>
        </CommentContainer>
      </div>
    );
  }
}

ChildComment.propTypes = {
  content: PropTypes.any,
  children: PropTypes.any,
  createdAt: PropTypes.string,
  authorId: PropTypes.string,
  ideaId: PropTypes.string,
  commentId: PropTypes.string,
};


const mapStateToProps = () => createStructuredSelector({
  comment: (state, { node }) => selectResourcesDomain('comments', node.id)(state),
});

const mergeProps = (stateProps) => {
  const { comment } = stateProps;
  // const { id } = ownProps.node;
  const attributes = comment.get('attributes');

  const content = attributes.get('body_multiloc');
  const createdAt = attributes.get('created_at');
  const authorId = comment.getIn(['relationships', 'author', 'data', 'id']);
  // const ideaId = comment.getIn(['relationships', 'idea', 'data', 'id']);
  return {
    comment,
    content,
    createdAt,
    authorId,
  };
};

export default preprocess(mapStateToProps, null, mergeProps)(ChildComment);
