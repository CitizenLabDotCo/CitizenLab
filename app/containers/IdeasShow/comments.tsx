import * as React from 'react';
import * as _ from 'lodash';
import * as Rx from 'rxjs/Rx';

// components
import ParentComment from './ParentComment';
import EditorForm from './EditorForm';
import Icon from 'components/UI/Icon';

// animations
import TransitionGroup from 'react-transition-group/TransitionGroup';
import CSSTransition from 'react-transition-group/CSSTransition';

// services
import { commentsForIdeaStream, commentStream, IComments, IComment } from 'services/comments';

// i18n
import { FormattedMessage } from 'react-intl';
import messages from './messages';

// style
import styled from 'styled-components';

const animationDuration = 500;
const animationEasing = `cubic-bezier(0.19, 1, 0.22, 1)`;

const Container = styled.div`
  will-change: opacity;

  &.background-enter {
    opacity: 0;

    &.background-enter-active {
      opacity: 1;
      transition: opacity ${animationDuration}ms ${animationEasing};
    }
  }
`;

const Title = styled.h2`
  font-size: 25px;
  font-weight: bold;
  margin-bottom: 30px;
`;

type Props = {
  ideaId: string;
};

type State = {
  comments: IComments | null;
  loading: boolean;
};

export default class Comments extends React.PureComponent<Props, State> {
  state: State;
  subscriptions: Rx.Subscription[];

  constructor() {
    super();
    this.state = {
      comments: null,
      loading: true
    };
    this.subscriptions = [];
  }

  componentWillMount() {
    const { ideaId } = this.props;
    const comments$ = commentsForIdeaStream(ideaId).observable;

    this.subscriptions = [
      comments$.switchMap((comments) => {
        if (comments && comments.data && comments.data.length > 0) {
          return Rx.Observable.combineLatest(
            comments.data.map(comment => commentStream(comment.id).observable)
          ).map(() => comments);
        }

        return Rx.Observable.of(null);
      }).delay(800).subscribe((comments) => {
        this.setState({ comments, loading: false });
      })
    ];
  }

  componentWillUnmount() {
    this.subscriptions.forEach(subscription => subscription.unsubscribe());
  }

  render() {
    const className = this.props['className'];
    const { ideaId } = this.props;
    const { comments, loading } = this.state;
    let commentsContainer: JSX.Element | null = null;

    if (!loading) {
      let commentsList: JSX.Element[] | null = null;

      if (comments && comments.data && comments.data.length > 0) {
        commentsList = comments.data.map(comment => (<ParentComment key={comment.id} commentId={comment.id} />));
      }

      commentsContainer = (
        <CSSTransition classNames="comments" timeout={animationDuration} exit={false}>
          <Container className={className}>
            <Title>
              <Icon name="comment outline" />
              <FormattedMessage {...messages.commentsTitle} />
            </Title>

            <EditorForm ideaId={ideaId} />

            {commentsList}
          </Container>
        </CSSTransition>
      );
    }

    return (
      <TransitionGroup>
        {commentsContainer}
      </TransitionGroup>
    );
  }
}
