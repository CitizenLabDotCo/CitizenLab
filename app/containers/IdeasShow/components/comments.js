/**
*
* CommentList
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
// import { Comment } from 'semantic-ui-react';
import ParentComment from './comments/ParentComment';
import selectIdeasShow, { makeSelectComments } from 'containers/IdeasShow/selectors';
import EditorForm from './common/editorForm';
import { publishCommentFork } from '../sagas';
import messages from '../messages';
// import MapChildren from './comments/mapChildren';


const Title = styled.h2`
  font-size: 25px;
  font-weight: bold;
`;

class CommentContainer extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { comments, className, storeCommentError, ideaId } = this.props;
    if (!comments) return null;
    return (
      <div className={className}>
        <Title><FormattedMessage {...messages.commentsTitle} /></Title>

        {storeCommentError && storeCommentError !== '' && <div>
          {storeCommentError}
        </div>}
        <EditorForm
          parentId={null}
          ideaId={ideaId}
          saga={publishCommentFork}
          onSuccess={this.closeEditor}
        />
        {comments.map((comment) =>
          <ParentComment key={comment.id} node={comment} />
        )}
      </div>
    );
  }
}

CommentContainer.propTypes = {
  comments: PropTypes.any,
  className: PropTypes.string,
  storeCommentError: PropTypes.string,
  ideaId: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  comments: makeSelectComments,
  storeCommentError: selectIdeasShow('storeCommentError'),
  resetEditorContent: selectIdeasShow('resetEditorContent'),
});

export default connect(mapStateToProps)(CommentContainer);
