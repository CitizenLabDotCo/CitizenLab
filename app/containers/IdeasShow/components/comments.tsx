import * as React from 'react';
import * as _ from 'lodash';
import * as Rx from 'rxjs/Rx';

// components
import ParentComment from './comments/ParentComment';
import EditorForm from './comments/EditorForm';
import Icon from 'components/UI/Icon';

// services
import { commentsForIdeaStream, commentStream, IComments, IComment } from 'services/comments';

// i18n
import { FormattedMessage } from 'react-intl';
import messages from '../messages';

// style
import styled from 'styled-components';

const Container = styled.div``;

const Title = styled.h2`
  font-size: 25px;
  font-weight: bold;
  margin-bottom: 30px;
`;

type Props = {
  ideaId: string;
};

type State = {
  comments: IComments | null;
  loading: boolean;
};

export default class Comments extends React.PureComponent<Props, State> {
  state: State;
  subscriptions: Rx.Subscription[];

  constructor() {
    super();
    this.state = {
      comments: null,
      loading: true
    };
    this.subscriptions = [];
  }

  componentWillMount() {
    const { ideaId } = this.props;
    const comments$ = commentsForIdeaStream(ideaId).observable;

    this.subscriptions = [
      comments$.switchMap((comments) => {
        if (comments && comments.data && comments.data.length > 0) {
          return Rx.Observable.combineLatest(
            comments.data.map(comment => commentStream(comment.id).observable)
          ).map(() => comments);
        }

        return Rx.Observable.of(null);
      }).delay(1000).subscribe((comments) => {
        this.setState({ comments, loading: false });
      })
    ];
  }

  componentWillUnmount() {
    this.subscriptions.forEach(subscription => subscription.unsubscribe());
  }

  render() {
    const className = this.props['className'];
    const { ideaId } = this.props;
    const { comments, loading } = this.state;

    if (!loading) {
      let commentsList: JSX.Element[] | null = null;

      if (comments && comments.data && comments.data.length > 0) {
        commentsList = comments.data.map(comment => (<ParentComment key={comment.id} commentId={comment.id} />));
      }

      return (
        <Container className={className}>
          <Title>
            <Icon name="comment outline" />
            <FormattedMessage {...messages.commentsTitle} />
          </Title>

          <EditorForm ideaId={ideaId} />

          {commentsList}
        </Container>
      );
    }

    return null;
  }
}
