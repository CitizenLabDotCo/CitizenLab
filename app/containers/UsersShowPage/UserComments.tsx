import React, { memo } from 'react';
import { isNilOrError } from 'utils/helperUtils';
import { adopt } from 'react-adopt';

// components
import IdeaCommentGroup from './IdeaCommentGroup';

// resources
import GetCommentsForUser, { GetCommentsForUserChildProps } from 'resources/GetCommentsForUser';
import { ICommentData } from 'services/comments';

// style
import styled, { withTheme } from 'styled-components';
import Button from 'components/UI/Button';

// intl
import { FormattedMessage } from 'utils/cl-intl';
import messages from './messages';
import { darken, rgba } from 'polished';
import { media, colors, fontSizes } from 'utils/styleUtils';
import GetAuthUser, { GetAuthUserChildProps } from 'resources/GetAuthUser';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin: auto;

  max-width: 760px;
`;

const Footer = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  margin-top: 30px;

  ${media.smallerThanMinTablet`
    flex-direction: column;
    align-items: stretch;
    margin-top: 0px;
  `}
`;

const MessageContainer = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${colors.label};
  font-size: ${fontSizes.medium}px;
  font-weight: 400;
`;

interface InputProps {
  userId: string;
}

interface DataProps {
  comments: GetCommentsForUserChildProps;
  authUser: GetAuthUserChildProps;
}

interface Props extends InputProps, DataProps {
  theme: any;
}

export const reducer = (acc: ICommentData[][], current: ICommentData) => {
  const accLen = acc.length;
  const lastArray = acc[accLen - 1];

  if (lastArray.length === 0) {
    return [[current]];
  }

  if (current.attributes.publication_status === 'published') {
    if (current.relationships.idea.data.id === lastArray[lastArray.length - 1].relationships.idea.data.id) {
      lastArray.push(current);
    } else {
      acc.push([current]);
    }
  }
  return acc;

};

export const UserComments = memo<Props>(({ comments, comments: { commentsList }, userId, theme, authUser }) => (
  !isNilOrError(commentsList) && commentsList.length > 0) ? (
    <Container className="e2e-profile-comments">
      {commentsList.reduce(reducer, [[]]).map(commentForIdea => (
        <IdeaCommentGroup
          key={commentForIdea[0].relationships.idea.data.id}
          ideaId={commentForIdea[0].relationships.idea.data.id}
          commentsForIdea={commentForIdea}
          userId={userId}
        />
      ))}
      {comments.hasMore &&
        <Footer>
          <Button
            onClick={comments.loadMore}
            processing={comments.loadingMore}
            icon="showMore"
            textColor={theme.colorText}
            textHoverColor={darken(0.1, theme.colorText)}
            bgColor={rgba(theme.colorText, 0.08)}
            bgHoverColor={rgba(theme.colorText, 0.12)}
            height="50px"
          >
            <FormattedMessage {...messages.loadMoreComments} />
          </Button>
        </Footer>
      }
    </Container>
  ) : (commentsList === undefined ? (
    <MessageContainer>
        <FormattedMessage {...messages.loadingComments} />
    </MessageContainer>
  ) : (commentsList === null || Array.isArray(commentsList) && commentsList.length === 0 ? (
    !isNilOrError(authUser) && userId === authUser.id ? (
      <MessageContainer>
        <FormattedMessage {...messages.noCommentsForYou} />
      </MessageContainer>
    ) : (
      <MessageContainer>
        <FormattedMessage {...messages.noCommentsForUser} />
      </MessageContainer>
    )
  ) : (
    <MessageContainer>
        <FormattedMessage {...messages.tryAgain} />
    </MessageContainer>
  ))
  )
);

const Data = adopt<DataProps, InputProps>({
  comments: ({ userId, render }) =>  <GetCommentsForUser userId={userId}>{render}</GetCommentsForUser>,
  authUser: <GetAuthUser />
});

const UserCommentsWithHocs = withTheme<Props, {}>(UserComments);

const WrappedUserComments = (inputProps: InputProps) => (
  <Data {...inputProps}>
    {dataProps => <UserCommentsWithHocs {...inputProps} {...dataProps} />}
  </Data>
);

export default WrappedUserComments;
