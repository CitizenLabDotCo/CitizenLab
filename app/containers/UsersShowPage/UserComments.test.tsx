// libraries
import React from 'react';
import { shallow } from 'enzyme';

// component to test
import { UserComments, reducer } from './UserComments';

// mock utilities
jest.mock('utils/cl-intl');
jest.mock('services/comments');
import { makeComments } from 'services/comments';

describe('<ConsentManager />', () => {

  it('reducer splits the array returned  by the back-end into arrays of comments belonging to the same idea', () => {
    const commentsAsReturned = makeComments([{ ideaId: 'idea1' }, { ideaId: 'idea1' }, { ideaId: 'idea1' }, { ideaId: 'idea2' }, { ideaId: 'idea2' }]);
    expect(commentsAsReturned.data.reduce(reducer, [[]]).map(arr => arr.map(comment => comment.relationships.idea.data.id))).toMatchSnapshot();
  });
});
