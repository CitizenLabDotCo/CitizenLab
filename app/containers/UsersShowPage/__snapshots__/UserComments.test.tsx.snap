// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserComments /> lets reducer splits the array returned  by the back-end into arrays of comments belonging to the same idea 1`] = `
Array [
  Array [
    "idea1",
    "idea1",
    "idea1",
  ],
  Array [
    "idea2",
    "idea2",
  ],
]
`;

exports[`<UserComments /> renders correctly when empty 1`] = `
<UserComments__MessageContainer>
  <FormattedMessage
    defaultMessage="This user has not posted any comment yet."
    id="app.containers.UsersShowPage.noCommentsForUser"
  />
</UserComments__MessageContainer>
`;

exports[`<UserComments /> renders correctly when empty and own profile 1`] = `
<UserComments__MessageContainer>
  <FormattedMessage
    defaultMessage="You have not posted any comment yet."
    id="app.containers.UsersShowPage.noCommentsForYou"
  />
</UserComments__MessageContainer>
`;

exports[`<UserComments /> renders correctly when error 1`] = `
<UserComments__MessageContainer>
  <FormattedMessage
    defaultMessage="An error has occured, please try again later."
    id="app.containers.UsersShowPage.tryAgain"
  />
</UserComments__MessageContainer>
`;

exports[`<UserComments /> renders correctly with actual comments 1`] = `
<UserComments__Container
  className="e2e-profile-comments"
>
  <WrappedIdeaComments
    commentsForIdea={
      Array [
        Object {
          "attributes": Object {
            "body_multiloc": Object {
              "en": "Just a comment",
            },
            "children_count": 0,
            "created_at": "2019-03-12T00: 00: 00.000Z",
            "downvotes_count": 0,
            "publication_status": "published",
            "updated_at": "2019-03-26T14: 32: 32.000Z",
            "upvotes_count": 0,
          },
          "id": "commentId",
          "relationships": Object {
            "author": Object {
              "data": Object {
                "id": "authorId",
                "type": "user",
              },
            },
            "idea": Object {
              "data": Object {
                "id": "idea1",
                "type": "idea",
              },
            },
            "parent": Object {
              "data": null,
            },
            "user_vote": Object {
              "data": null,
            },
          },
          "type": "comment",
        },
      ]
    }
    ideaId="idea1"
    key="idea1"
  />
  <WrappedIdeaComments
    commentsForIdea={
      Array [
        Object {
          "attributes": Object {
            "body_multiloc": Object {
              "en": "Just a comment",
            },
            "children_count": 0,
            "created_at": "2019-03-12T00: 00: 00.000Z",
            "downvotes_count": 0,
            "publication_status": "published",
            "updated_at": "2019-03-26T14: 32: 32.000Z",
            "upvotes_count": 0,
          },
          "id": "commentId",
          "relationships": Object {
            "author": Object {
              "data": Object {
                "id": "authorId",
                "type": "user",
              },
            },
            "idea": Object {
              "data": Object {
                "id": "idea2",
                "type": "idea",
              },
            },
            "parent": Object {
              "data": null,
            },
            "user_vote": Object {
              "data": null,
            },
          },
          "type": "comment",
        },
        Object {
          "attributes": Object {
            "body_multiloc": Object {
              "en": "Just a comment",
            },
            "children_count": 0,
            "created_at": "2019-03-12T00: 00: 00.000Z",
            "downvotes_count": 0,
            "publication_status": "published",
            "updated_at": "2019-03-26T14: 32: 32.000Z",
            "upvotes_count": 0,
          },
          "id": "commentId",
          "relationships": Object {
            "author": Object {
              "data": Object {
                "id": "authorId",
                "type": "user",
              },
            },
            "idea": Object {
              "data": Object {
                "id": "idea2",
                "type": "idea",
              },
            },
            "parent": Object {
              "data": null,
            },
            "user_vote": Object {
              "data": null,
            },
          },
          "type": "comment",
        },
      ]
    }
    ideaId="idea2"
    key="idea2"
  />
</UserComments__Container>
`;

exports[`<UserComments /> renders correctly with actual comments and more 1`] = `
<WithTheme(Button)
  bgColor="rgba(230,143,81,0.08)"
  bgHoverColor="rgba(230,143,81,0.12)"
  height="50px"
  icon="showMore"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  textColor="#e68f51"
  textHoverColor="#e07224"
>
  <FormattedMessage
    defaultMessage="Load more comments"
    id="app.containers.UsersShowPage.loadMoreComments"
  />
</WithTheme(Button)>
`;
